P1_BMI_Stats.R

```diff
@@ -588,25 +587,90 @@ bmi_basefile <- bmi_data %>%
 # those published in financial year 2017/18.
 saveRDS(bmi_basefile, paste0(host_folder, "BMI_data_0102_1718.rds"))
 
+#614,752 obs.
+
+bmi_basefile <- readRDS(paste0(host_folder, "BMI_data_0102_1718.rds"))
 
 
 ### Health board analysis
 
 # SQ
 # create population file from the GRO mid year population estimates
 # of five year olds in each HB and for all participating boards
-hb_pop_estimates <-
+hb_pop_estimates <- readRDS(paste0(
+  lookupFolder, "/Unicode/Populations/Estimates/HB2019_pop_est_1981_2018.rds"))
+
+hb_pop_estimates <- hb_pop_estimates %>% 
+  filter(age == 5) %>%
+  filter(year >= 2000 & year <=2018) %>% 
+  mutate(schlyr_exam = case_when(year == 2001 ~ "0102", 
+                                 year == 2002 ~ "0203",
+                                 year == 2003 ~ "0304",
+                                 year == 2004 ~ "0405",
+                                 year == 2005 ~ "0506",
+                                 year == 2006 ~ "0607",
+                                 year == 2007 ~ "0708",
+                                 year == 2008 ~ "0809",
+                                 year == 2009 ~ "0910",
+                                 year == 2010 ~ "1011",
+                                 year == 2011 ~ "1112",
+                                 year == 2012 ~ "1213",
+                                 year == 2013 ~ "1314",
+                                 year == 2014 ~ "1415",
+                                 year == 2015 ~ "1516",
+                                 year == 2016 ~ "1617",
+                                 year == 2017 ~ "1718",
+                                 year == 2018 ~ "1819")) %>% 
+  # do we need to pipe to functions?
+  # call the function for creating HB cypher
+  apply_hb_cypher %>%
+  # do we pipe after functions?
+
+# call the function for selecing the relevant year for each board
+apply_hb_year(HB = 'F', ey = 102)
+apply_hb_year(HB = 'L', ey = 102)
+apply_hb_year(HB = 'S', ey = 102)
+apply_hb_year(HB = 'T', ey = 203)
+apply_hb_year(HB = 'W', ey = 304)
+apply_hb_year(HB = 'Y', ey = 405)
+apply_hb_year(HB = 'V', ey = 506)
+apply_hb_year(HB = 'G', ey = 607)
+apply_hb_year(HB = 'A', ey = 708)
+apply_hb_year(HB = 'H', ey = 809)
+apply_hb_year(HB = 'Z', ey = 809)
+apply_hb_year(HB = 'N', ey = 910)
+apply_hb_year(HB = 'R', ey = 1011)
+
+# create totals for individual hb and all 
+# participating boards (for hb_pop_estimates)
+# Board level
+hb_pop_estimates <- rbind(hb_pop_estimates %>% 
```

Comment:
Need to specify sum function to be applied to pop. Also to use HB2019 and ungroup afterwords, otherwise it passed all values as lists to two rows of data:
         hb_pop_estimates <- rbind(hb_pop_estimates %>% 
                            group_by(HB2019, schlyr_exam)%>%
                            summarise(pop = sum(pop)) %>% ungroup,
                          # Scotland level (all participating boards)
                          hb_pop_estimates %>% group_by(schlyr_exam) %>%
                            summarise(pop = sum(pop)) %>%
                            mutate(HB2019 = "Total")  %>% ungroup)

Reply:
updated.